### PR TITLE
FIX: #5640 Bad price for predefine product

### PR DESCRIPTION
**Pull request by olsesacl**
This error persists in Dolibarr 4.0.

I edit files that are in different directory in Dolibarr 4.0.
If I have to create another pull request to that let me know and I think other with editions for that version

Delete 

> $cancel     = GETPOST('cancel','alpha');

It hasn't any funcion really, because in loop if else use for example:

> else if ($action == 'updateline' && $user->rights->ficheinter->creer && **GETPOST('cancel')**)

### CODE REVIEW

File: htdocs/comm/propal.php [hunk 1]
```diff
@@ -117,7 +116,6 @@
 
 if (empty($reshook))
 {
-	if ($cancel) $action = '';
```
Comment (eldy) [inline htdocs/comm/propal.php:120]: Can you keep this.
For future, we must change code to follow this rule that is much secured than adding a test && ! GETPOST('cancel') on each following test

File: htdocs/commande/card.php [hunk 2]
```diff
@@ -109,7 +109,6 @@
 
 if (empty($reshook))
 {
-	if ($cancel) $action='';
```
Comment (eldy) [inline htdocs/commande/card.php:112]: Can you keep this.
For future, we must change code to follow this rule that is much secured than adding a test && ! GETPOST('cancel') on each following test

**Review by eldy — CHANGES_REQUESTED** [review]

**Comment by eldy** [comment]
Yes, can you also submit a fix for 4.0 (but keep usage of cancel that is cleared at beginning of "action area".

**Comment by olsesacl** [comment]
@eldy , changed correctly?

**Comment by eldy** [comment]
No.
The ligne
if ($cancel) $action='';
that is a securit to clear vaue of action if we click on a cancel has still disappears.

**Comment by eldy** [comment]
What about version 4.0.3 ?

**Comment by eldy** [comment]
I have some reason to think this may be fixed or no more applicable into more recent version. Does this bug still occurs with last stable version ?
If yes, please reopen the ticket and mention the recent version you tried with...

**Pull request closed without merging**